aloha-h2o/src/main/scala/com/eharmony/aloha/models/h2o/H2oModel.scala

```diff
@@ -0,0 +1,249 @@
+//       the VW model, then we can get the same guarantees.
+final case class H2oModel[-A, +B](
+    modelId: ModelIdentity,
+    modelSource: ModelSource,
+    featureNames: sci.IndexedSeq[String],
+    featureFunctions: sci.IndexedSeq[FeatureFunction[A]],
+    numMissingThreshold: Option[Int] = None)(implicit private[this] val scb: ScoreConverter[B])
+  extends BaseModel[A, B]
+     with Logging {
+
+  // Because H2o's RowData object is essentially a Map of String to Object, we unapply the wrapper
+  // and throw away the type information on the function return type.  We have type safety because
+  // FeatureFunction is sealed (ADT).
+  @transient private[this] lazy val lazyAnyRefFF = featureFunctions map {
+    case DoubleFeatureFunction(f) => f
+    case StringFeatureFunction(f) => f
+  }
+
+  @transient private[this] lazy val h2oPredictor: RowData => Either[IllConditioned, B] = {
+    val sourceFile = new java.io.File(modelSource.localVfs.descriptor)
+    val p = getH2oPredictor(sourceFile, _.fromFile).get
+    if (modelSource.shouldDelete)
+      Try[Unit] { sourceFile.delete() }
+    p
+  }
+
+  // Force initialization of lazy vals.
+  require(lazyAnyRefFF != null)
+  require(h2oPredictor != null)
+
+  override private[aloha] def getScore(a: A)(implicit audit: Boolean): (ModelOutput[B], Option[Score]) = {
+    val f = constructFeatures(a)
+    if (!f.missingOk)
+      failureDueToMissing(f.missing)
+    else
+      try {
+        predict(f)
+      } catch {
+        // We know about this specifically from the H2o documentation.
+        case e: PredictUnknownCategoricalLevelException                => handleBadCategorical(e, f)
+        case e: IllegalArgumentException if isCategoricalMissing(e, f) => handleMissingCategorical(e, f)
+      }
+  }
+
+  protected[this] def predict(f: Features[RowData])(implicit audit: Boolean) =
+    h2oPredictor(f.features).fold(ill => failure(Seq(ill.errorMsg), getMissingVariables(f.missing)),
+                                  s   => success(s))
+
+  /**
+    */
+
+  /**
+    * ''Attempt'' to determine if a categorical was missing in the h2o model.
+    *
+    * Currently (3.6.0.3), h2o generated model says: "" when a categorical value is not supplied.
+    * This is determined from inspecting the generated H2o model code so it's likely brittle and subject
+    * to change but its better than throwing an IllegalArgumentException with no diagnostics information
+    * when there is missing data in a categorical variable.
+    * @param e exception thrown by h2o
+    * @param f the data passed in.
+    * @return whether to attempt to recover.  Don't attempt to recover unless a string-based feature appears to
+    *         be missing.  This is so that we can diagnose when the model will fail every time.
+    */
+  protected[this] def isCategoricalMissing(e: IllegalArgumentException, f: Features[RowData]): Boolean =
```

Comment:
Look here.  I tried to make this forgiving as to the error message but I wanted to include the check for the feature specification of a string-based feature.  See scaladoc comments above.